### PR TITLE
Callable prototypes

### PR DESCRIPTION
**Pull request by dktapps**
This is a working implementation, although a bit rough around the edges.

A new toggle `checkMissingCallablePrototype` is added, disabled by default (as per @ondrejmirtes request).

### CODE REVIEW

File: src/Rules/MissingTypehintCheck.php [hunk 1]
```diff
@@ -133,4 +139,27 @@ public function getNonGenericObjectTypesWithGenericClass(Type $type): array
...
+	{
+		if (!$this->checkMissingCallablePrototype) {
+			return [];
+		}
+
+		$result = [];
+		TypeTraverser::map($type, static function (Type $type, callable $traverse) use (&$result): Type {
+			if (
+				($type instanceof CallableType && $type->isCommonCallable()) ||
+				($type instanceof ObjectType && $type->getClassName() === \Closure::class)) {
```
Comment (dktapps) [inline src/Rules/MissingTypehintCheck.php:156]: This is the only thing I'm not happy with. I think there should be a better, more generalized way to do this, but if there is one it's not apparent to me.

Comment (ondrejmirtes) [reply src/Rules/MissingTypehintCheck.php:156]: This is probably fine... There's `Type::isCallable()` and `Type::getCallableParametersAcceptors()` but there's no way to extract whether it's general callable or already specified callable from that.

So as long as this place is synchronized with `TypeNodeResolver::resolveCallableTypeNode()`, we're fine.

**Comment by ondrejmirtes** [comment]
I've already added a few commits locally, I'm taking over the PR, thanks.

**Comment by ondrejmirtes** [comment]
In master. Thank you.

**Comment by ondrejmirtes** [comment]
@dktapps Please submit a documentation change that explains this setting somewhere here: https://phpstan.org/config-reference#stricter-analysis Thanks.

**Comment by dktapps** [comment]
Awesome, thanks! Will attend to the docs as soon as I get home.

**Pull request closed without merging**